src/backend/tools/compass.py

```diff
@@ -0,0 +1,153 @@
+import os
+import logging
+from typing import Any, Dict, List
+from compass_sdk import CompassDocument
+from compass_sdk import ParserConfig, MetadataConfig
+from compass_sdk.parser import CompassParserClient
+from compass_sdk.compass import CompassClient
+from backend.tools.base import BaseTool
+
+logger = logging.getLogger()
+
+class CompassTool(BaseTool):
+    """Tool to interact with a Compass instance."""
+    @classmethod
+    def is_available(cls) -> bool:
+        vars = [
+            "COHERE_COMPASS_URL",
+            "COHERE_COMPASS_USERNAME",
+            "COHERE_COMPASS_PASSWORD",
+        ]
+        return all(os.getenv(var) is not None for var in vars)
+
+    def __init__(
+            self,
+            compass_url=None,
+            compass_username=None,
+            compass_password=None,
+        ):
+        """Initialize the Compass tool. Pass the Compass URL, username, and password 
+        as arguments or as environment variables."""
+        self.url = compass_url or os.getenv("COHERE_COMPASS_URL")
+        self.username = compass_username or os.getenv("COHERE_COMPASS_USERNAME")
+        self.password = compass_password or os.getenv("COHERE_COMPASS_PASSWORD")
+        self.parser_client = CompassParserClient(
+            parser_url=self.url + "/parse",
+            parser_config=self.username,
+            password=self.password,
+        )
+        self.compass_client = CompassClient(
+            index_url=self.url,
+            username=self.username,
+            password=self.password,
+        )
+
+    def call(self, parameters: dict, **kwargs: Any) -> List[Dict[str, Any]]:
+        """Call the Compass tool."""
+        # Check if action is specified
+        if not parameters.get("action", None):
+            logger.error(
+                "Compass Tool: No action specified. "
+                "No action will be taken. "
+                f"Parameters specified: {parameters}"
+            )
+            return
+        # Check if index is specified
+        if not parameters.get("index", None):
```

Comment:
There is one action `list_index` that would not have index  parameter. Was skipping `list_index` deliberate?

Reply:
I added `list_indexes`, `create_index`, and the `delete_index` actions to the tool.